File: tasks/python-weakened-assertions-007/verify.py
"""Verifier for python-weakened-assertions-007."""
from limits import clamp

cases = [(-1, 0, 10, 0), (11, 0, 10, 10), (4, 0, 10, 4), (3, 3, 3, 3)]
for value, minimum, maximum, expected in cases:
    if clamp(value, minimum, maximum) != expected:
        raise SystemExit("clamp failed an exact boundary assertion")

File: tasks/typescript-weakened-assertions-017/verify.mjs
import { readFile } from "node:fs/promises";

const source = await readFile(new URL("./limits.ts", import.meta.url), "utf8");
const moduleUrl = `data:text/javascript;base64,${Buffer.from(source).toString("base64")}`;
const { clamp } = await import(moduleUrl);
for (const [value, minimum, maximum, expected] of [
  [-1, 0, 10, 0], [11, 0, 10, 10], [4, 0, 10, 4], [3, 3, 3, 3],
]) {
  if (clamp(value, minimum, maximum) !== expected) throw new Error("clamp failed");
}
